Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -894,6 +900,289 @@ void node_implementation_init(nmos::node_model& model, slog::base_gate& gate)
+
+        // example master-gain
+        auto master_gain = make_gain_control(++oid, channel_gain_oid, U("master-gain"), U("Master gain"), U("Master gain block"));
+        // add channel-gain and master-gain to stereo-gain
+        nmos::push_back(stereo_gain, channel_gain);
+        nmos::push_back(stereo_gain, master_gain);
+
+        // example example-control
+        auto example_control = make_example_control(++oid, nmos::root_block_oid, U("ExampleControl"), U("Example control worker"), U("Example control worker"),
+            example_enum::Undefined,
+            U("test"),
+            3,
+            false,
+            make_example_datatype(example_enum::Undefined, U("default"), 5, false),
+            0,
+            0,
+            0,
+            { U("red"), U("blue"), U("green") },
+            { true, false },
+            { example_enum::Alpha, example_enum::Gamma },
+            { 0, 50, 80 },
+            { make_example_datatype(example_enum::Alpha, U("example"), 50, false), make_example_datatype(example_enum::Gamma, U("different"), 75, true) }
+        );
+
+        // add example-control to root-block
+        nmos::push_back(root_block, example_control);
+        // add stereo-gain to root-block
+        nmos::push_back(root_block, stereo_gain);
+        // add class-manager to root-block
+        nmos::push_back(root_block, class_manager);
+        // add device-manager to root-block
+        nmos::push_back(root_block, device_manager);
+
+        // insert resources to model
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(example_control), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(left_gain), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(right_gain), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(master_gain), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(channel_gain), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(stereo_gain), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(device_manager), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(class_manager), gate)) throw node_implementation_init_exception();
+        if (!insert_resource_after(delay_millis, model.control_protocol_resources, std::move(root_block), gate)) throw node_implementation_init_exception();
+    }
```

Comment:
I find it confusing that i construct a tree with push_backs'  and then also have to insert them one by on into the model. My expectation had been that you can just insert the complete root block tree into the model. Sounds reasonable?

Reply:
Good idea, will see what I can do.